File: app/src/main/java/gateway/hitrontech/com/encryption/utils/RandomStringUtils.java
package gateway.hitrontech.com.encryption.utils;

import gateway.hitrontech.com.encryption.bean.LanguageUnicode;
import java.util.Random;

public class RandomStringUtils {

  private LanguageUnicode unicode;

  private RandomStringUtils(LanguageUnicode unicode) {
    this.unicode = unicode;
  }

  public static RandomStringUtils getInstance(LanguageUnicode unicode) {
    return new RandomStringUtils(unicode);
  }

  public static void main(String[] args) {
    RandomStringUtils utils = RandomStringUtils.getInstance(Constants.JAPAN);
    utils.getRandomString(15);

  }

  public String getRandomString(int length) {
    int measure = unicode.getUnicodeEnd() - unicode.getUnicodeStart();
    Random random = new Random();
    StringBuilder stringBuilder = new StringBuilder();

    for (int i = 0; i < length; i++) {
      int tmp = random.nextInt(measure);
      int result = (unicode.getUnicodeStart() + tmp);
      stringBuilder.append((char) result);
      // System.out.println(stringBuilder.toString());

    }

    return stringBuilder.toString();
  }
}

File: app/src/main/java/gateway/hitrontech/com/encryption/utils/RandomStringUtils.kt
package gateway.hitrontech.com.encryption.utils

import gateway.hitrontech.com.encryption.bean.LanguageUnicode
import java.util.*

class RandomStringUtils private constructor(private val unicode: LanguageUnicode) {

    fun getRandomString(length: Int): String {
        val measure = unicode.unicodeEnd - unicode.unicodeStart
        val random = Random()
        val stringBuilder = StringBuilder()

        for (i in 0 until length) {
            val tmp = random.nextInt(measure)
            val result = unicode.unicodeStart + tmp
            stringBuilder.append(result.toChar())
            // System.out.println(stringBuilder.toString());

        }

        return stringBuilder.toString()
    }

    companion object {

        fun getInstance(unicode: LanguageUnicode): RandomStringUtils {
            return RandomStringUtils(unicode)
        }

        @JvmStatic
        fun main() {
            val utils = RandomStringUtils.getInstance(Constants.JAPAN)
            utils.getRandomString(15)

        }
    }
}
